### PR TITLE
Various cleanups of universe apis

### PR DESCRIPTION
**Pull request by SkySkimmer**
See commit messages.

### CODE REVIEW

**Review by ppedrot — COMMENTED** [review]
I have a mixed feeling about this PR. While most of it is a very welcome cleanup, I am not so convinced by most of the moves of functions from upper layers into the kernel. There is no reason to inflate it with stuff that is only used by universe unification...

**Comment by SkySkimmer** [comment]
I'll have another look when I rebase then.

**Comment by ppedrot** [comment]
Maybe to be more precise: my main concern is `universes_of_constr` which is somewhat nonsense.

File: kernel/typeops.ml [hunk 1]
```diff
@@ -432,21 +432,8 @@ and execute_array env = Array.map (execute env)
...
-    match kind c with
-    | Const (_c, u) ->
-       LSet.fold LSet.add (Instance.levels u) s
-    | Ind ((_mind,_), u) | Construct (((_mind,_),_), u) ->
-       LSet.fold LSet.add (Instance.levels u) s
-    | Sort u when not (Sorts.is_small u) ->
-      let u = Sorts.univ_of_sort u in
-      LSet.fold LSet.add (Universe.levels u) s
-    | _ -> Constr.fold aux s c
-  in aux LSet.empty c
```
Comment (SkySkimmer) [inline kernel/typeops.ml]: universes_of_constr was already in the kernel, see this part of the diff

Comment (ppedrot) [reply kernel/typeops.ml]: Bah, I missed that. I guess that's OK then.

**Review by ppedrot — APPROVED** [review]

File: kernel/typeops.ml (hunk 1 above)
Comment (mattam82) [reply kernel/typeops.ml]: It is used to check the universes are declared mainly, what's wrong with that?

Comment (ppedrot) [reply kernel/typeops.ml]: I read it too quickly and thought that was the old version that was randomly picking universes declared in the constants. The current version is sane indeed.

**Comment by ppedrot** [comment]
@mattam82, @maximedenes If that's fine with you guys I'll merge that.

**Review by mattam82 — APPROVED** [review]
Nice cleanups indeed, nothing to add. I think the move of globref manipulations to the kernel is harmless and useful.

**Comment by ppedrot** [comment]
CI failure spurious, about to merge.

**Pull request merged**